ndscan/experiment/fragment.py

```diff
@@ -641,13 +705,107 @@ def get_default_analyses(self) -> Iterable[DefaultAnalysis]:
         return []
 
 
+def _skip_common_prefix(target: list, reference: list) -> list:
+    i = 0
+    while i < len(target) and i < len(reference) and target[i] == reference[i]:
+        i += 1
+    return target[i:]
+
+
+class AggregateExpFragment(ExpFragment):
+    r"""Combines multiple :class:`ExpFragment`\ s by executing them one after each other
+    each time :meth:`run_once` is called.
+
+    For ``@kernel`` fragments, the top-level :meth:`run_once` is implemented on the core
+    device as well to avoid costly kernel recompilations at each point.
+    """
+    def build_fragment(self, exp_fragments: List[ExpFragment]) -> None:
+        """
+        :param exp_fragments: The "child" fragments to execute. The fragments will be
+            run in the given order. No special treatment is given to the
+            ``{host,device}_{setup,cleanup}()`` methods, which by default will just be
+            executed using the default recursive behaviour.
+        """
+        if not exp_fragments:
+            raise ValueError("At least one child ExpFragment should be given")
+        self._exp_fragments = exp_fragments
+
+        is_kernels = [is_kernel(e.run_once) for e in exp_fragments]
+        if all(is_kernels):
+            self.run_once = kernel(self.run_once)
```

Comment:
This doesn't compile:
```
root:While compiling <repository>/...
/ndscan/ndscan/experiment/fragment.py:756:9-756:17: error: cannot unify ()->NoneType delay('a) with (self:'b)->'c delay('d)
    def run_once(self) -> None:
        ^^^^^^^^               
/ndscan/ndscan/experiment/fragment.py:756:9-756:17: note: expression of type ()->NoneType delay('a)
    def run_once(self) -> None:
        ^^^^^^^^               
```
I tried the following fix, which worked:
```python
self.run_once = self._krun_once
```
with an explicit kernel method
```python
    @kernel
    def _krun_once(self) -> None:
        return self._run_once_impl(self)
```